nservicebus/upgrades/6to7/moving-msmq-subscriptions.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Migrating Msmq subscription messages
+reviewed: 2017-10-05
+component: MsmqTransport
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 6
+ - 7
+---
+
+When using Msmq Subscription storage in NServiceBus Versions 5.x and Versions 6.x, when a subscription queue is not configured, the subscription storage queue automatically defaulted to `NServiceBus.Subscriptions`. However, as this queue name was not endpoint specific, endpoints deployed to the same server could potentially share the same subscriptions queue which is undesirable. 
+
+To rectify this, starting from Versions 1.0 and above of NServiceBus.Transport.Msmq, if a queue called `NServiceBus.Subscriptions` is detected, an exception will be thrown to prevent potential loss of messages. 
+
+To move messages from the `NServiceBus.Subscriptions` queue to the new queue, refer to the instructions outlined below.
+
+## Create the subscriptions queue
+
+Create a transactional queue called [EndpointName].Subscriptions and substitute the actual name for the endpoint. This can be done by using the Powershell scripts, using [native tools or other tools like QueueExplorer](/transports/msmq/viewing-message-content-in-msmq.md#windows-native-tools). If creating queues manually using UI tools, ensure that the queues are marked, "transactional". 
+
+### Using Powershell Script
+
+Use the `CreateQueue` function that's part of `CreateQueues.ps1`. This powershell script, comes packaged with the NServiceBus.Transport.Msmq NuGet package. Browse to the lib folder to locate the scripts, for example, `%USERPROFILE%\.nuget\packages\nservicebus.transport.msmq\1.0.0\lib\net452\Scripts`. 
+
+2. [Load the `CreateQueues.ps1` powershell script](https://technet.microsoft.com/en-us/library/bb613481.aspx) and run the `CreateQueue` function as shown below:
+
+```
+    # CreateQueue -QueueName "EndpointName.Subscriptions" -Account $env:USERNAME
+```
+
+
+## Move the subscription messages 
+
+Once the new queue is created, use a tool like [QueueExplorer](http://www.cogin.com/mq/index.php) to locate the messages in the `NServiceBus.Subscriptions` queue and move them to the newly created subscriptions queue. If the `NServiceBus.Subscriptions`queue was being shared among multiple endpoints, select only the messages intended for the endpoint that is being upgraded to NServiceBus.Transport.Msmq package. Identify which messages need to be moved by inspecting the message body and looking for the event information and also by inspecting the information in the `LABEL` column.
```

Comment:
Let's be explicit in saying the subcriber queue name is found in the `LABEL`. _Information_ is too fuzzy.